### PR TITLE
fix(releases): delete older than X

### PR DESCRIPTION
**Pull request by s0up4200**
Resolves a bug in the `DeleteOlder` function where recent 24-hour data wasn't deleted when set to delete everything. We now correctly set the `olderThanTimestamp` to a future date when duration is zero, ensuring complete deletion of all records.

### CODE REVIEW

**Comment by s0up4200** [comment]
Another option is to do this I guess:

```go
olderThanTimestamp := time.Now().UTC().Add(-time.Duration(duration) * time.Hour)

if duration == 0 {
	olderThanTimestamp = time.Unix(1<<63-62135596801, 999999999)
}
```

File: internal/database/release.go [hunk 1]
```diff
@@ -594,7 +594,14 @@ func (repo *ReleaseRepo) DeleteOlder(ctx context.Context, duration int) error {
...
+
+	if duration == 0 {
+		olderThanTimestamp = time.Now().UTC().AddDate(100, 0, 0) // set future date to delete everything
+	} else {
+		olderThanTimestamp = time.Now().UTC().Add(-time.Duration(duration) * time.Hour)
+	}
+
 	log.Debug().Msgf("Deleting releases older than: %v", olderThanTimestamp)
 
 	result, err := tx.ExecContext(ctx, `DELETE FROM "release" WHERE timestamp < $1`, olderThanTimestamp)
```
Comment (KyleSanderson) [inline internal/database/release.go:607]: This still isn't correct. It needs to go through the process that was done for limits.

Comment (s0up4200) [reply internal/database/release.go:607]: Can you elaborate in soup terms?

Comment (KyleSanderson) [reply internal/database/release.go:607]: https://github.com/autobrr/autobrr/pull/883/files all those casts are needed unfortunately.

**Review by zze0s — APPROVED** [review]
Made some changes and now it's good to go!

**Pull request merged**